### PR TITLE
ci: ubuntu 18.04 is deprecated upgrade to ubuntu 22.04

### PR DESCRIPTION
**Pull request by subhamkrai**
**Description of your changes:**
Look at individual commit

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
@travisn instead of backporting all the commits what about if we just BP commit which we want like we want ubuntu 20.04 support and supporting k8s 17.17 right? Just to avoid any overlook issue.

**Comment by subhamkrai** [comment]
has anyone noticed golangci lint error in master?

**Comment by satoru-takeuchi** [comment]
@subhamkrai I've never seen this error.

**Comment by subhamkrai** [comment]
> @subhamkrai I've never seen this error.

I'll try something to get it working

**Comment by parth-gr** [comment]
Still, I see failure in Canary integration tests / raw-disk and CRDs gen / crds-gen are they intermittent?

**Comment by subhamkrai** [comment]
> Still, I see failure in Canary integration tests / raw-disk and CRDs gen / crds-gen are they intermittent?

raw disk failure looks like intermittent  and crds gen ci was not failing till today will check

**Comment by subhamkrai** [comment]
@travisn ready for review and bp?

File: .github/workflows/integration-test-config-older-k8s/action.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Cluster Setup
```
Comment (travisn) [inline .github/workflows/integration-test-config-older-k8s/action.yaml:1]: Didn't we decide that using the different older/newer config didn't help much? Or does this still help with k8s 1.17/1.18?

Comment (subhamkrai) [reply .github/workflows/integration-test-config-older-k8s/action.yaml:1]: I thought we decided to keep it in 1.9 for 1.18

Comment (subhamkrai) [reply .github/workflows/integration-test-config-older-k8s/action.yaml:1]: also, origin pr was merged in 1.10 branch too.

**Review by travisn — APPROVED** [review]

**Pull request merged**